esmvaltool/cmorizers/data/formatters/datasets/esacci_sst.py

```diff
@@ -56,8 +56,9 @@ def extract_variable(var_info, raw_info, attrs, year):
     try:
         cube = iris.load_cube(raw_info['file'], constraint)
     except iris.exceptions.ConstraintMismatchError:
-        raise ValueError(f"No data available for variable {rawvar}"
-                         f"and year {year}")
+        logger.error("No data available for variable %s and year %s",
+                     rawvar, year)
+        raise
```

Comment:
I suppose you got something about a missing `from` and re-raising here. I suppose that this version is one way of dealing with it, but just for completeness sake, you *can* use the `ValueError` as follows:
```suggestion
    except iris.exceptions.ConstraintMismatchError as constraint_error:
        raise ValueError(f"No data available for variable {rawvar}"
                         f"and year {year}") from constraint_error
```
I somehow like this version better, but I have no strong preference, so you choose.

Reply:
See also https://stackoverflow.com/questions/24752395/python-raise-from-usage

Reply:
Thanks! I hesitated between the 2 options and gave `logger.error` a chance but I see your point.